### PR TITLE
All items on page

### PR DESCRIPTION
**Pull request by dexter0175**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,43 +1,78 @@
...
-  res.header("Access-Control-Allow-Headers", "x-access-token, Origin, Content-Type, Accept");
+  res.header(
+    "Access-Control-Allow-Headers",
+    "x-access-token, Origin, Content-Type, Accept"
+  );
   next();
 });
 
-require('./onliner-listener');
+//require('./onliner-listener');
```
Comment (ops1ops) [inline server.js]: uncomment

File: server.js [hunk 2]
```diff
@@ -1,43 +1,78 @@
-require('dotenv').config();
-const express = require('express');
-const db = require('./db');
+require("dotenv").config();
```
Comment (ops1ops) [inline server.js]: use single quoted string

File: server.js [hunk 3]
```diff
@@ -1,43 +1,78 @@
...
+
+//     return item;
+//   });
+
+//   console.log(items[4].history); - OK. mapping works perfectly
+
+//   viewModel.items = items;
+//   return res.render("home", viewModel );
+//   => in EJS - items not mapped => WTF???
+//   fucking magic.
```
Comment (ops1ops) [inline server.js]: no commented code

File: views/home.ejs [hunk 4]
```diff
@@ -1,23 +1,87 @@
...
-        const q = <%= test%>
-        console.log(q)
-    </script>
-</body>
-</html>
\ No newline at end of file
+    <script src="https://canvasjs.com/assets/script/canvasjs.min.js"></script>
+  </head>
+  <body>
+    <!-- <div id="chartContainer" style="height: 300px; width: 100%;"></div> -->
```
Comment (ops1ops) [inline views/home.ejs]: no commented code

File: views/home.ejs [hunk 5]
```diff
@@ -1,23 +1,87 @@
...
+		item.history = item.history.map(el => ({
+			x: new Date(el.createdAt),
+			y: Number.parseFloat(el.price)
+		}));
+		return item;
+	 });
+
+	 
+      items.forEach(item => {
+        let chart = getChart(item.name, item.history);
```
Comment (ops1ops) [inline views/home.ejs]: const

File: views/home.ejs [hunk 6]
```diff
@@ -1,23 +1,87 @@
...
\ No newline at end of file
+    <script src="https://canvasjs.com/assets/script/canvasjs.min.js"></script>
+  </head>
+  <body>
+    <!-- <div id="chartContainer" style="height: 300px; width: 100%;"></div> -->
+  </body>
+  <script>
+    window.onload = function() {
+      let items = JSON.parse('<%- JSON.stringify(items) %>');
+	  const getChart = getChartFactory();
```
Comment (ops1ops) [inline views/home.ejs]: extra spaces

**Comment by dexter0175** [comment]
wtf

File: views/home.ejs [hunk 7]
```diff
@@ -1,23 +1,87 @@
...
+	  items =  items.map(item => {
+		if(!item.history){
+			item.history = [];
+			return item
+		}
```
Comment (ops1ops) [inline views/home.ejs]: not needed

File: views/home.ejs [hunk 8]
```diff
@@ -1,23 +1,87 @@
...
+		item.history = item.history.map(el => ({
```
Comment (ops1ops) [inline views/home.ejs]: use 2 spaces tab

File: views/home.ejs [hunk 9]
```diff
@@ -1,23 +1,87 @@
...
+	 });
+
+	 
+      items.forEach(item => {
+        let chart = getChart(item.name, item.history);
+        chart.render();
+      });
+    };
+
+    function getChartFactory() {
```
Comment (ops1ops) [inline views/home.ejs]: arrow function

File: views/home.ejs [hunk 10]
```diff
@@ -1,23 +1,87 @@
...
-</body>
-</html>
\ No newline at end of file
+    <script src="https://canvasjs.com/assets/script/canvasjs.min.js"></script>
+  </head>
+  <body>
+    <!-- <div id="chartContainer" style="height: 300px; width: 100%;"></div> -->
+  </body>
+  <script>
+    window.onload = function() {
```
Comment (ops1ops) [inline views/home.ejs]: arrow function

File: views/home.ejs [hunk 11]
```diff
@@ -1,23 +1,87 @@
...
+    };
+
+    function getChartFactory() {
+      let id = 0;
+
+      return (name, data) => {
+		const CHART_ID  = "chart-" + id 
+		id++;
+
+        let body = document.querySelector("body");
```
Comment (ops1ops) [inline views/home.ejs]: const

File: views/home.ejs [hunk 12]
```diff
@@ -1,23 +1,87 @@
...
+        let chartContainer = document.createElement("div");
```
Comment (ops1ops) [inline views/home.ejs]: const

File: server.js [hunk 13]
```diff
@@ -1,43 +1,78 @@
...
+
+//   console.log(items[4].history); - OK. mapping works perfectly
+
+//   viewModel.items = items;
+//   return res.render("home", viewModel );
+//   => in EJS - items not mapped => WTF???
+//   fucking magic. 
+  viewModel.items = items;
+
+  return res.render("home", viewModel );
```
Comment (ops1ops) [inline server.js]: try `return res.render("home", { items } );`

File: views/home.ejs [hunk 14]
```diff
@@ -1,23 +1,87 @@
...
+    <script src="https://canvasjs.com/assets/script/canvasjs.min.js"></script>
+  </head>
+  <body>
+    <!-- <div id="chartContainer" style="height: 300px; width: 100%;"></div> -->
+  </body>
+  <script>
+    window.onload = function() {
+      let items = JSON.parse('<%- JSON.stringify(items) %>');
+	  const getChart = getChartFactory();
+	  items =  items.map(item => {
```
Comment (ops1ops) [inline views/home.ejs]: extra space

File: views/home.ejs [hunk 15]
```diff
@@ -1,23 +1,87 @@
...
+		if(!item.history){
+			item.history = [];
+			return item
+		}
+		item.history = item.history.map(el => ({
+			x: new Date(el.createdAt),
+			y: Number.parseFloat(el.price)
+		}));
+		return item;
```
Comment (ops1ops) [inline views/home.ejs]: empty string before return

File: server.js (hunk 13 above)
Comment (dexter0175) [reply server.js]: suddenly, it doesn't work

**Pull request merged**